homeassistant/components/camera/webrtc.py

```diff
@@ -129,6 +210,85 @@ async def _async_refresh_providers(hass: HomeAssistant) -> None:
     )
 
 
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "camera/webrtc/offer",
+        vol.Required("entity_id"): cv.entity_id,
+        vol.Required("offer"): str,
+    }
+)
+@websocket_api.async_response
+async def ws_webrtc_offer(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """Handle the signal path for a WebRTC stream.
+
+    This signal path is used to route the offer created by the client to the
+    camera device through the integration for negotiation on initial setup.
+    The ws endpoint returns an subscription id, where ice candidates and the
+    final answer will be returned.
+    The actual streaming is handled entirely between the client and camera device.
+
+    Async friendly.
+    """
+    entity_id = msg["entity_id"]
+    offer = msg["offer"]
+    camera = get_camera_from_entity_id(hass, entity_id)
+    if camera.frontend_stream_type != StreamType.WEB_RTC:
+        connection.send_error(
+            msg["id"],
+            "web_rtc_offer_failed",
+            (
+                "Camera does not support WebRTC,"
+                f" frontend_stream_type={camera.frontend_stream_type}"
+            ),
+        )
+        return
+
+    session_id = uuid.uuid4().hex
+    connection.subscriptions[msg["id"]] = partial(
+        camera.close_webrtc_session, session_id
+    )
+
+    # Send subscription id to client
+    connection.send_result(msg["id"], {"session_id": session_id})
+
+    @callback
+    def send_message(message: WebRTCMessages | dict[str, Any]) -> None:
+        """Push a value to websocket."""
+        value = message if isinstance(message, dict) else message.to_dict()
+        connection.send_message(
+            websocket_api.event_message(
+                msg["id"],
+                value,
+            )
+        )
+
+    send_message(WebRTCSessionId(session_id))
+
+    if camera.supports_async_webrtc_offer:
+        await camera.async_handle_web_rtc_offer(offer, session_id, send_message)
```

Comment:
You are not sending a websocket response when the camera supports async offers. This means that the frontend will never unsubscribe.

You can also turn this into a guard clause
```suggestion
    if camera.supports_async_webrtc_offer:
        await camera.async_handle_web_rtc_offer(offer, session_id, send_message)
        return
```